File: frontend/src/app/shared/models/Meme.ts
export class Meme{
  id: number;
  title: string;
  imageblob: Blob;
  likes: number;
  dislikes: number;
  imageSrc: string;

  constructor(title: string, image: Blob, id: number){
    this.title = title;
    this.imageblob = image;
    this.id = id
    this.likes = 0;
    this.dislikes = 0;
    this.imageSrc = ""
  }
}

File: backend/src/main/java/com/example/memegenerator/data/entity/Meme.java
package com.example.memegenerator.data.entity;

import lombok.Getter;
import lombok.Setter;

import javax.persistence.*;
import javax.validation.constraints.NotNull;

import java.util.HashSet;
import java.util.Set;

@Entity
@Getter
@Setter
public class Meme extends BaseEntity {

    @Column(name = "title", nullable = false)
    @NotNull
    public String title;

    @Column(name = "description")
    public String description;

    @Column(name = "image", nullable = false)
    public byte[] imageblob;

    @Column(name = "likes")
    public int likes;

    @Column(name = "dislikes")
    public int dislikes;

    @ManyToOne
    @JoinColumn(name = "imageid")
    public Image image;

    @ManyToOne
    @JoinColumn(name = "userid", nullable = false)
    public User user;

    @ManyToOne
    @JoinColumn(name = "categoryid", nullable = false)
    public Category category;

    @ManyToMany(cascade = { CascadeType.ALL })
    @JoinTable(name = "meme_tags", joinColumns = { @JoinColumn(name = "memeid") }, inverseJoinColumns = {
            @JoinColumn(name = "tagid") })
    public Set<Tag> tags = new HashSet<>();
}